Modules/ExtendedPlayerControl.cs

```diff
@@ -1375,7 +1372,7 @@ public static bool KnowSubRoleTarget(PlayerControl seer, PlayerControl target)
             else if (seer.Is(CustomRoles.Egoist) && target.Is(CustomRoles.Egoist) && Egoist.ImpEgoistVisibalToAllies.GetBool())
                 return true;
         }
-        else if (Admirer.IsEnable && Admirer.KnowRole(seer, target)) return true;
+        else if (CustomRoles.Admirer.GetRoleClass().IsEnable && Admirer.KnowRole(seer, target)) return true;
```

Comment:
use the native RoleExist() or IsEnable() customroles extension, bcuz this creates a new object.

Reply:
Yeah this code does not work at all and will always return false since it creates a bew instance which isn't initialized....
Use `Main.PlayerStates.Any(x => x.Value.Role is Admirer {IsEnable: true})` (I typed this out manually so it may have errors)